tracer/src/Datadog.Trace/Util/RequestDataHelper.cs

```diff
@@ -0,0 +1,122 @@
+// <copyright file="RequestDataHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+#if NETFRAMEWORK
+using System.Collections.Specialized;
+using System.Web;
+using Datadog.Trace.Logging;
+#else
+using Microsoft.AspNetCore.Http;
+#endif
+
+#nullable enable
+
+namespace Datadog.Trace.Util;
+
+internal static class RequestDataHelper
+{
+#if NETFRAMEWORK
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RequestDataHelper));
+
+    // Get the querystring from a HttpRequest
+    internal static NameValueCollection? GetQueryString(HttpRequest request)
+    {
+        try
+        {
+            return request.QueryString;
+        }
+        catch (HttpRequestValidationException)
+        {
+            Log.Debug("Error reading QueryString from the request.");
+            return null;
+        }
+    }
+#else
+    internal static QueryString GetQueryString(HttpRequest request)
+    {
+        return request.QueryString;
+    }
+#endif
+
+#if NETFRAMEWORK
+    // Get the cookies from a HttpRequest
+    internal static HttpCookieCollection? GetCookies(HttpRequest request)
+    {
+        try
+        {
+            return request.Cookies;
+        }
+        catch (HttpRequestValidationException)
+        {
+            Log.Debug("Error reading Cookies from the request.");
+            return null;
+        }
+    }
+#else
+    internal static IRequestCookieCollection GetCookies(HttpRequest request)
+    {
+        return request.Cookies;
+    }
+#endif
+
+#if NETFRAMEWORK
+    // Get the cookies from a HttpRequest
+    internal static NameValueCollection? GetHeaders(HttpRequest request)
+    {
+        try
+        {
+            return request.Headers;
```

Comment:
I suspect we call this lots of otherplaces too 😕 For example our scope extraction would presumably also trigger this? Do we need to use the helper in that case as well, or is that safe somehow? 🤔

Reply:
You don't have this kind of error since 2.54
Taking a close [look at the logs](https://app.datadoghq.com/logs?query=source%3Adotnet%20%40tracer_version%3A2.54%2A%20-%2AAppsec%2A%20%2AHttpRequestValidationException%2A%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AgAAAZGS9eifLX0U6QAAAAAAAAAYAAAAAEFaR1M5ZWlmQUFBUzU0MUpoam10MHdBQQAAACQAAAAAMDE5MTkzMDItN2VkMi00NWIwLTkzZTQtODI2YzIzZTQ1ZjQ5&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1720177260603&to_ts=1720782060603&live=true), you used to have it in TracingHttpModule.cs, but now you capture all the exceptions.

                if (tracer.InternalActiveScope == null)
                {
                    try
                    {
                        // extract propagated http headers
                        headers = httpRequest.Headers.Wrap();
                        propagatedContext = SpanContextPropagator.Instance.Extract(headers.Value);
                    }
                    catch (Exception ex)
                    {
                        Log.Error(ex, "Error extracting propagated HTTP headers.");
                    }
                }
                
I think that in this piece of code, it would be better to use the RequestDataHelper. I will fix it and take a look closer in other places.

Reply:
I have made some updates to on the tracinghttpmodule.cs I have not identified other places that use httprequest in net framework where it could happen. The logs also don't show any HttpRequestValidationException out of the already modified classes. I was not sure about AspNetMvcIntegration for instance, but I tried with request after calling ValidateInput and it seems to be fine.